server/routes/application/application.router.js

```diff
@@ -0,0 +1,31 @@
+const express = require("express");
+
+const applicationRouter = express.Router();
+
+const {
+    httpHandleApplicationCreation,
+    httpHandleGetApplicationById,
+    httpHandleApplicationSearch,
+    httpHandleApplicationUpdate,
+    httpHandleApplicationDelete
+} = require("./application.controller");
+
+
+//Create a new application
+applicationRouter.post('/application', httpHandleApplicationCreation);
+
+//Update an existing application
+applicationRouter.patch('/application/:id', httpHandleApplicationUpdate)
+
+//Get an application by id
+applicationRouter.get('/application/:id', httpHandleGetApplicationById)
+
+
+//Get an application by search query
+applicationRouter.get('/application/search', httpHandleApplicationSearch)
```

Comment:
This may seem weird but can you rename the endpoint to /applications/search (with a plural) ? I think that may fix your bug....

Reply:
:)
:)

It seems that woks on plural.
Going to test a bit more.
Many thanks